psh/test-ls-rootfs.py

```diff
@@ -0,0 +1,76 @@
+# Phoenix-RTOS
+#
+# phoenix-rtos-tests
+#
+# additional psh ls command test for targets with root file system
+#
+# Copyright 2022 Phoenix Systems
+# Author: Damian Loewnau
+#
+# This file is part of Phoenix-RTOS.
+#
+# %LICENSE%
+#
+
+import psh.tools.psh as psh
+
+
+def assert_ls_pshcmds(p):
+    psh_cmds = psh.get_commands(p)
+
+    psh_cmd_pattern = r'(\x1b\[\S*)('
+    for psh_cmd in psh_cmds:
+        if psh_cmd != psh_cmds[0]:
+            psh_cmd_pattern += r'|'
+        psh_cmd_pattern += psh_cmd
+    psh_cmd_pattern += r')((\s)|(\x1b\[\S*))'
+
+    cmd = 'ls bin'
+    listed_psh_cmds = []
+    p.sendline(cmd)
+    p.expect(cmd)
+    idx = 0
+    while True:
+        idx = p.expect([psh.PROMPT, psh_cmd_pattern])
+        if idx == 1:
+            groups = p.match.groups()
+            if groups[1] not in listed_psh_cmds:
+                listed_psh_cmds.append(groups[1])
+        elif idx == 0:
+            break
+    msg = (f"The number of psh commands listed in bin is incorrect: {len(listed_psh_cmds)} instead of {len(psh_cmds)}."
+           + f"\nListed psh commands: {sorted(listed_psh_cmds)}"
+           + f"\nPsh commands got from help: {sorted(psh_cmds)}")
+
+    assert len(listed_psh_cmds) == len(psh_cmds), msg
+
+
+def assert_ls_t(p):
+    msg_date = "Wrong output when setting date with `date` command!"
+    msg_touch = "Wrong output when creating file with `touch` command!"
+    msg_mkdir = "Wrong output when creating directory with `mkdir` command!"
```

Comment:
I think about removing the second part of msg statement, because called command is printed in reader buffer, like here:
![Screenshot from 2022-03-22 14-12-12](https://user-images.githubusercontent.com/77304431/159489976-06a73370-909c-440c-8a5d-e4ed28621006.png)

Reply:
Most important thing to include in output from tested device. Everything else can be deduced from test source code. Its nice to have send input and expected output. Additional message should not be needed in most cases.